### DISCUSSION TITLE
[d3-org-chart] exports a conflicting namespace with d3

### DISCUSSION THREAD

**Discussion by hinzed1127** [DefinitelyTyped/DefinitelyTyped#63266] [Issues with a @types package]
The types for `d3` and `d3-org-chart` both export "d3" as a a namespace (see [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3/index.d.ts#L13) and [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3-org-chart/index.d.ts#L9)). This is causing a resolution conflict when using both in the same project, like so:

A component with the following imports/declarations:
```typescript
import { select } from 'd3-selection'
import { zoom as d3Zoom } from 'd3-zoom'
// ........
// ........
const svg = select('#canvas') as d3.Selection<
    HTMLCanvasElement,
    unknown,
    HTMLElement,
    undefined
  >
// ........
// ........
  function zoomed(event: d3.D3ZoomEvent<SVGSVGElement, unknown>) {
// ........
// ........
}
```
results in the following TypeScript errors (abridged error message with the relevant bits):
```
Error: Map.tsx(63,39): error TS2694: Namespace '"{project_dir}/node_modules/@types/d3-org-chart/index"' has no exported member 'Selection'.
Error: Map.tsx([17]): error TS2694: Namespace '"{project_dir}/node_modules/@types/d3-org-chart/index"' has no exported member 'D3ZoomEvent'.
```

Removing `export as namespace d3;` from `@types/d3-org-chart` solves this issue. It also seems sensible to me that only the type definitions for the full and official `d3` package have this kind of namespace. Would that be an appropriate fix that I could open a PR for?

PS It seems like this is a similar issue as #57377, but I've only personally encountered and troubleshooted the problem as described here.

[a comment by an automated account is not shown]

**Comment by domdomegg** [comment]
Happy to accept a PR on this, provided the tests still pass and the types still all work as a library :)

**Reply by hinzed1127** [reply]
Thanks! PR [created](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63289). There was one other small fix I added too: just a method that previously wasn't part of the public API.
